<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from model.xacro                    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="HolonomicUrdf">
  <link name="base_footprint"/>
  <link name="base_link">
    <inertial>
      <mass value="100"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="28.166666666666664" ixy="0.0" ixz="0.0" iyy="28.166666666666664" iyz="0.0" izz="54.166666666666664"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.5 0.5 0.1"/>
      </geometry>
      <material name="blue">
        <color rgba="0.0 0.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
  </joint>
  <link name="wheel_base_link">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="233333.33333333334" ixy="0.0" ixz="0.0" iyy="233333.33333333334" iyz="0.0" izz="450000.0"/>
    </inertial>
  </link>
  <joint name="wheel_base_joint" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="wheel_link">
    <inertial>
      <mass value="1e-07"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0000083333333336" ixy="0.0" ixz="0.0" iyy="1.0000083333333336" iyz="0.0" izz="2.0000000000000004"/>
    </inertial>
    <collision>
      <geometry>
        <cylinder length="0.001" radius="0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="wheel_joint" type="continuous">
    <parent link="wheel_base_link"/>
    <child link="wheel_link"/>
    <origin rpy="-1.57 0 0" xyz="0 0 0.15"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="caster1_link">
    <inertial>
      <mass value="0.3"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1470000.0000000005" ixy="0.0" ixz="0.0" iyy="1470000.0000000005" iyz="0.0" izz="1470000.0000000005"/>
    </inertial>
    <visual>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <material name="green">
        <color rgba="0.0 1.0 0.0 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="caster1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.25000000000000006 0.25000000000000006 -0.015"/>
    <parent link="base_link"/>
    <child link="caster1_link"/>
  </joint>
  <link name="caster2_link">
    <inertial>
      <mass value="0.3"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1470000.0000000005" ixy="0.0" ixz="0.0" iyy="1470000.0000000005" iyz="0.0" izz="1470000.0000000005"/>
    </inertial>
    <visual>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <material name="green">
        <color rgba="0.0 1.0 0.0 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="caster2_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.25000000000000006 0.25000000000000006 -0.015"/>
    <parent link="base_link"/>
    <child link="caster2_link"/>
  </joint>
  <link name="caster3_link">
    <inertial>
      <mass value="0.3"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1470000.0000000005" ixy="0.0" ixz="0.0" iyy="1470000.0000000005" iyz="0.0" izz="1470000.0000000005"/>
    </inertial>
    <visual>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <material name="green">
        <color rgba="0.0 1.0 0.0 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="caster3_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.25000000000000006 -0.25000000000000006 -0.015"/>
    <parent link="base_link"/>
    <child link="caster3_link"/>
  </joint>
  <link name="caster4_link">
    <inertial>
      <mass value="0.3"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1470000.0000000005" ixy="0.0" ixz="0.0" iyy="1470000.0000000005" iyz="0.0" izz="1470000.0000000005"/>
    </inertial>
    <visual>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <material name="green">
        <color rgba="0.0 1.0 0.0 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="caster4_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.25000000000000006 -0.25000000000000006 -0.015"/>
    <parent link="base_link"/>
    <child link="caster4_link"/>
  </joint>
  <link name="base_lidar_link">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="8.333333333333335e-06" ixy="0.0" ixz="0.0" iyy="8.333333333333335e-06" iyz="0.0" izz="1.2500000000000002e-05"/>
    </inertial>
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="base_lidar_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_lidar_link"/>
    <origin rpy="0 0 0" xyz="0.15 0 0.011"/>
  </joint>
  <gazebo reference="base_lidar_link">
    <sensor name="gpu_lidar" type="gpu_lidar">
      <topic>/scan</topic>
      <ignition_frame_id>base_lidar_link</ignition_frame_id>
      <update_rate>10</update_rate>
      <ray>
        <scan>
          <horizontal>
            <samples>270</samples>
            <resolution>1</resolution>
            <min_angle>-2.356194490192345</min_angle>
            <max_angle>2.356194490192345</max_angle>
          </horizontal>
        </scan>
        <range>
          <min>0.01</min>
          <max>20</max>
        </range>
        <noise>
          <type>gaussian</type>
          <mean>0.0</mean>
          <stddev>0.005</stddev>
        </noise>
      </ray>
      <always_on>true</always_on>
      <visualize>true</visualize>
    </sensor>
  </gazebo>
  <gazebo>
    <!-- Global odometer -->
    <plugin filename="ignition-gazebo-odometry-publisher-system" name="ignition::gazebo::systems::OdometryPublisher">
      <odom_publish_frequency>50</odom_publish_frequency>
      <odom_topic>/odom</odom_topic>
      <odom_frame>odom</odom_frame>
      <robot_base_frame>base_footprint</robot_base_frame>
      <tf_topic>/odom/tf</tf_topic>
    </plugin>
    <!-- Joint state publisher -->
    <plugin filename="libignition-gazebo-joint-state-publisher-system.so" name="ignition::gazebo::systems::JointStatePublisher">
      <topic>joint_states</topic>
    </plugin>
    <!-- Holonomic plugin -->
    <plugin filename="libHolonomicPlugin.so" name="holonomic_sim::HolonomicPlugin">
      <wheel_joint>wheel_joint</wheel_joint>
      <base_joint>wheel_base_joint</base_joint>
    </plugin>
  </gazebo>
</robot>
